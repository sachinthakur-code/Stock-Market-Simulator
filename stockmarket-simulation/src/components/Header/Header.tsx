import * as React from 'react';
import { Navbar } from 'react-bootstrap';
import * as ReactDOM from 'react-dom';
import appRoutes from '../../routes/routes';

interface HeaderProps {
}

interface HeaderState {
    sidebarExists: boolean;
}

export class Header extends React.Component<HeaderProps, HeaderState> {

    constructor( props: HeaderProps ) {
        super( props );
        this.mobileSidebarToggle = this.mobileSidebarToggle.bind( this );
        this.state = {
            sidebarExists: false
        };
    }

    // tslint:disable-next-line no-any
    mobileSidebarToggle( e: any ) {
        if (this.state.sidebarExists === false) {
            this.setState( {
                               sidebarExists: true
                           } );

        }
        e.preventDefault();
        document.documentElement.classList.toggle( 'nav-open' );
        const node = document.createElement( 'div' );
        node.id = 'bodyClick';
        node.onclick = function () {
            // tslint:disable-next-line no-any
            ReactDOM.findDOMNode(this as any)!.parentElement!.removeChild( this as any );
            document.documentElement.classList.toggle( 'nav-open' );
        };
        document.body.appendChild( node );
    }

    getBrand() {
        const routeIndex = appRoutes.findIndex( r => window.location.pathname.indexOf( r.path ) !== -1 );
        const appRoute = appRoutes[routeIndex];

        if (appRoute && appRoute.name) {
            return appRoute.name;
        }
        return 'Stockmarket Simulation';
    }

    render() {
        return (
            <Navbar fluid={true} >
                <Navbar.Header>
                    <Navbar.Brand>
                        <a href="http://localhost/login/logout.php">Logout</a>
                    </Navbar.Brand>
                    <Navbar.Toggle onClick={this.mobileSidebarToggle}/>
                </Navbar.Header>
            </Navbar>
        );
    }
}
