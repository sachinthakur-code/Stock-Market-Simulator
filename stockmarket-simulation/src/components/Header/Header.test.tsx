import * as React from 'react';
import { configure, mount, shallow } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { Header } from './Header';

configure( { adapter: new ReactSixteenAdapter() } );

describe( '<Header />', () => {
    it( 'shallow Snapshot matches', () => {
        const component = shallow( <Header /> );
        expect( component ).toMatchSnapshot();
    } );

    it( 'Header should not explode', () => {
        const component = mount( <Header /> );
        expect( component ).not.toBeNull();
    } );
} );
