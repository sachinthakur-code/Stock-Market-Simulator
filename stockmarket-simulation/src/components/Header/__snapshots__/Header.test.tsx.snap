// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shallow Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "bsStyle": "default",
      "children": <NavbarHeader>
        <NavbarBrand>
                <a
                        href="#"
                >
                        Depot
                </a>
        </NavbarBrand>
        <NavbarToggle
                onClick={[Function]}
        />
</NavbarHeader>,
      "fluid": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NavbarBrand>
            <a
                        href="#"
            >
                        Depot
            </a>
</NavbarBrand>,
          <NavbarToggle
            onClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <a
              href="#"
>
              Depot
</a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Depot",
              "href": "#",
            },
            "ref": null,
            "rendered": "Depot",
            "type": "a",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Brand": [Function],
      "Collapse": [Function],
      "ControlledComponent": [Function],
      "Form": [Function],
      "Header": [Function],
      "Link": [Function],
      "STYLES": Array [
        "default",
        "inverse",
      ],
      "Text": [Function],
      "Toggle": [Function],
      "defaultProps": Object {
        "bsStyle": "default",
      },
      "deferControlTo": [Function],
      "propTypes": Object {
        "bsStyle": [Function],
        "defaultExpanded": [Function],
        "expanded": [Function],
        "innerRef": [Function],
      },
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "bsStyle": "default",
        "children": <NavbarHeader>
          <NavbarBrand>
                    <a
                              href="#"
                    >
                              Depot
                    </a>
          </NavbarBrand>
          <NavbarToggle
                    onClick={[Function]}
          />
</NavbarHeader>,
        "fluid": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <a
                            href="#"
              >
                            Depot
              </a>
</NavbarBrand>,
            <NavbarToggle
              onClick={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                href="#"
>
                Depot
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Depot",
                "href": "#",
              },
              "ref": null,
              "rendered": "Depot",
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Brand": [Function],
        "Collapse": [Function],
        "ControlledComponent": [Function],
        "Form": [Function],
        "Header": [Function],
        "Link": [Function],
        "STYLES": Array [
          "default",
          "inverse",
        ],
        "Text": [Function],
        "Toggle": [Function],
        "defaultProps": Object {
          "bsStyle": "default",
        },
        "deferControlTo": [Function],
        "propTypes": Object {
          "bsStyle": [Function],
          "defaultExpanded": [Function],
          "expanded": [Function],
          "innerRef": [Function],
        },
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
